frame/contracts/src/gas.rs

```diff
@@ -147,6 +142,14 @@ impl<T: Trait> GasMeter<T> {
 		}
 	}
 
+	// Account for not fully used gas.
+	//
+	// This can be used after dispatching a runtime call to refund gas that was not
+	// used by the dispatchable.
+	pub fn refund(&mut self, gas: Gas) {
+		self.gas_left = (self.gas_left + gas).max(self.gas_limit);
```

Comment:
Do we guarantee that `gas_left + gas` does not overflow? Wouldn't it be safer to use `saturating_add` here?

Reply:
No we do not not. Good catch. Fixed it.